### PR TITLE
add MemRefDereferencing interface

### PR DESCRIPTION
**Pull request by adstraw**

### CODE REVIEW

File: mlir/include/mlir/Dialect/Affine/IR/AffineOps.td [hunk 1]
```diff
@@ -698,7 +699,7 @@ def AffineParallelOp : Affine_Op<"parallel",
   let hasFolder = 1;
 }
 
-def AffinePrefetchOp : Affine_Op<"prefetch"> {
+def AffinePrefetchOp : Affine_Op<"prefetch", [DeclareOpInterfaceMethods<MemRefDereferencingInterface>]> {
```
Comment (flaub) [inline mlir/include/mlir/Dialect/Affine/IR/AffineOps.td:702]: Does this exceed 80 cols?

**Review by flaub — APPROVED** [review]
LGTM

Clearly better to open the set via an interface over having closed isa's. 

However, I do wonder if we should allow the ability for ops to support multiple memrefs. Like a 'gemm' op may have 3 memrefs.

**Comment by flaub** [comment]
It makes me wonder if we could somehow use the side-effect traits to obtain memrefs.

File: mlir/include/mlir/Dialect/Affine/IR/AffineOps.td (hunk 1 above)
Comment (adstraw) [reply mlir/include/mlir/Dialect/Affine/IR/AffineOps.td]: Done.

**Comment by adstraw** [comment]
> LGTM
> 
> Clearly better to open the set via an interface over having closed isa's.
> 
> However, I do wonder if we should allow the ability for ops to support multiple memrefs. Like a 'gemm' op may have 3 memrefs.

pxa.gemm will need to implement a custom `getAffineMapAttrForMemRef` method

**Pull request merged**